### PR TITLE
Test move add remove

### PR DESCRIPTION
**Pull request by dantleech**
Replaces #22

### CODE REVIEW

File: Tests/Functional/CompositeRepositoryTest.php [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
- *
- * (c) 2011-2016 Symfony CMF
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Symfony\Cmf\Bundle\ResourceBundle\Tests\Functional;
-
-class CompositeRepositoryTest extends PhpcrRepositoryTestCase
```
Comment (dantleech) [inline Tests/Functional/CompositeRepositoryTest.php]: Puli 1.0 no longer supports the Composite repository (it was broken anyway).

File: Tests/Functional/PhpcrRepositoryTestCase.php [hunk 2]
```diff
@@ -11,32 +11,126 @@
...
+
+        $this->session->getNode($path);
+    }
 
-        $this->repositoryRegistry = $this->container->get('cmf_resource.registry');
+    public function provideRemove()
+    {
+        return [
+            ['/foo', 2],
+            ['/bar', 1],
```
Comment (dantleech) [inline Tests/Functional/PhpcrRepositoryTestCase.php]: Add test for globbed query

File: Tests/Resources/app/config/resource.yml [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
             type: doctrine_phpcr_odm
             basepath: /test
 
+        test_repository_phpcr:
+            type: doctrine_phpcr
+            basepath: /test
+
         articles:
             type: doctrine_phpcr_odm
             basepath: /cmf/articles
```
Comment (dantleech) [inline Tests/Resources/app/config/resource.yml]: is this repository used?

File: Tests/Resources/app/config/resource.yml [hunk 4]
```diff
@@ -15,5 +19,5 @@ cmf_resource:
         stuff:
             type: composite
```
Comment (dantleech) [inline Tests/Resources/app/config/resource.yml]: remove

File: Tests/Functional/CompositeRepositoryTest.php (hunk 1 above)
Comment (dantleech) [inline Tests/Functional/CompositeRepositoryTest.php]: Am fully removing all the Composite stuff in #32

File: Tests/Functional/PhpcrRepositoryTestCase.php [hunk 5]
```diff
@@ -11,32 +11,138 @@
...
+        }
+    }
 
-        $this->repositoryRegistry = $this->container->get('cmf_resource.registry');
+    public function provideRemove()
+    {
+        return [
+            ['/node-1', [ '/test/node-1' ]],
+            ['/node-2', [ '/test/node-2' ]],
+            ['/*', [ '/node-1', '/test/node-2' ]],
```
Comment (dantleech) [inline Tests/Functional/PhpcrRepositoryTestCase.php]: /test/node-1

File: Tests/Functional/RepositoryTestCase.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * @var RepositoryRegistryInterface
+     */
+    protected $repositoryRegistry;
+
+    /**
+     * @var SessionInterface
+     */
+    protected $session;
```
Comment (dantleech) [inline Tests/Functional/RepositoryTestCase.php]: move to PHPCR test case

File: composer.json [hunk 7]
```diff
@@ -12,8 +12,8 @@
     "minimum-stability": "dev",
     "require": {
         "php": "^5.5.9|^7.0",
-        "symfony/framework-bundle": "^2.7|3.*",
-        "symfony-cmf/resource": "^1.0"
+        "symfony/framework-bundle": "^2.3|3.*",
```
Comment (dantleech) [inline composer.json]: 2.7

**Comment by dantleech** [comment]
@electricmaxxx @wouterj good to review.

**Comment by dantleech** [comment]
@WouterJ @ElectricMaxxx will merge this tomorrow if no objections.

File: Tests/Functional/PhpcrRepositoryTestCase.php [hunk 8]
```diff
@@ -11,32 +11,137 @@
...
+ */
+abstract class PhpcrRepositoryTestCase extends RepositoryTestCase
 {
-    protected $dm;
+    /**
+     * @var SessionInterface
+     */
+    protected $session;
 
     public function setUp()
```
Comment (WouterJ) [inline Tests/Functional/PhpcrRepositoryTestCase.php]: protected

**Comment by WouterJ** [comment]
Not really sure why this is in the bundle and not the component.

**Comment by dantleech** [comment]
yep in know. the reason is because its easier to bootstrap here and we implicitly test the DI config, but I would prefer them to be in the component now.

but out of scope here as all the functional tests are in this bundle.

**Comment by WouterJ** [comment]
Anyway, once the public/protected thing is fixed, 👍 for merging this. (extraction can be done later)

**Comment by dantleech** [comment]
Fixed the protected stuff. Merging. We are getting there :)

**Pull request merged**